### PR TITLE
cli: deprecate the cockroach connect functionality

### PR DESCRIPTION
**Pull request by andrewbaptist**
Cockroach connect is going to be removed in a future release. It was never completely finished and there is no public documentation about the feature.

Epic: none

Release note (cli change): Deprecate the cockroach connect functionality.

### CODE REVIEW

**Comment by andrewbaptist** [comment]
@shralex - I'm not sure who should review this. This simply adds the deprecation warning to these commands which are removed in https://github.com/cockroachdb/cockroach/pull/113893 in 24.1. We want to backport to 23.2, so ideally it gets merged by tomorrow (prior to beta).

@williamkulju This change simply adds the deprecation warning to the command for users on 23.2.

File: pkg/cli/connect.go
Comment (erikgrinaker) [inline pkg/cli/connect.go:42]:
```suggestion
	Deprecated: "self-generated certificates are no longer supported, use '--insecure' instead",
```

Do we really want to encourage people to use `--insecure`?

**Review by erikgrinaker — APPROVED** [review]
> Should I target this against master first and then backport?

May as well.

I don't know anything about this functionality, or whether it makes sense to remove it, but assuming it does then the deprecation itself LGTM.

**Review by andrewbaptist — COMMENTED** [review]
This was a discussion between @shralex and @petermattis that we want to deprecate and remove it. I'll wait for @shralex to approve also that we are deprecating it. and I'll rebase against master instead.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114153#-:-NitaJehBE9flqMD-lgm:b-wbf8o4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @erikgrinaker, @shralex, and @williamkulju)

___
*[`pkg/cli/connect.go` line 42 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/114153#gh-1389459773:-NitaPGhERqqshoKOgsE:b-5681ze) ([raw file](https://github.com/cockroachdb/cockroach/blob/424b69ae086b4cc75a76a0b0ea46574a11ffc120/pkg/cli/connect.go#L42)):*
<details><summary><i>Previously, erikgrinaker (Erik Grinaker) wrote…</i></summary><blockquote>

```suggestion
	Deprecated: "self-generated certificates are no longer supported, use '--insecure' instead",
```

Do we really want to encourage people to use `--insecure`?
</blockquote></details>

This was what `cockroach connect` was meant to address. It is used in cases where there isn't an "external" place to generate certs, I could update the text to say "use a CA generated certificate for production cluster or --insecure for test clusters".

**Review by erikgrinaker — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114153)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @shralex and @williamkulju)

___
*[`pkg/cli/connect.go` line 42 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/114153#gh-1389459773:-NitbBp2BWyDvaj_hUxJ:bb7422a) ([raw file](https://github.com/cockroachdb/cockroach/blob/424b69ae086b4cc75a76a0b0ea46574a11ffc120/pkg/cli/connect.go#L42)):*
<details><summary><i>Previously, andrewbaptist (Andrew Baptist) wrote…</i></summary><blockquote>

This was what `cockroach connect` was meant to address. It is used in cases where there isn't an "external" place to generate certs, I could update the text to say "use a CA generated certificate for production cluster or --insecure for test clusters".
</blockquote></details>

Sgtm

**Comment by andrewbaptist** [comment]
@williamkulju 

If you attempt to run ./cockroach, the `connect` option will now be hidden, however it can still be called by calling `./cockroach connect` Here is output from both commands:

```
% ./cockroach
CockroachDB command-line interface and server.

Usage:
  cockroach [command]

Available Commands:
  start             start a node in a multi-node cluster
  start-single-node start a single-node cluster
  init              initialize a cluster
  cert              create ca, node, and client certs
  sql               open a sql shell
  statement-diag    commands for managing statement diagnostics bundles
  auth-session      log in and out of HTTP sessions
  node              list, inspect, drain or remove nodes

  nodelocal         upload and delete nodelocal files
  userfile          upload, list and delete user scoped files
  import            import a db or table from a local PGDUMP or MYSQLDUMP file
  demo              open a demo sql shell
  convert-url       convert a SQL connection string for use with various client drivers
  gen               generate auxiliary files
  version           output version information
  debug             debugging commands
  sqlfmt            format SQL statements
  workload          generators for data and query loads
  help              Help about any command
...
```

```
% ./cockroach connect
Command "connect" is deprecated, self generated certificates are no longer supported functionality.
Use a CA generated certificate for production cluster or --insecure for test clusters instead.

Usage:
  cockroach connect [command] [flags]

Flags:
  -h, --help   help for connect

Global Flags:
...
```

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114153#-:-Nitm6sKADU4mMQRKJDi:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @erikgrinaker, @shralex, and @williamkulju)

**Review by williamkulju — APPROVED** [review]

**Comment by andrewbaptist** [comment]
bors r+

TFTR!

**Pull request merged**
